### ISSUE TITLE
Consider adding a `@JunitParam` or something similar which would allow Arquillian to ignore the method parameter

### ISSUE THREAD

**Issue by jamezp** [arquillian/arquillian-core#610]
##### Issue Overview

Both Arquillian and JUnit 5 allow for parameters to be resolved for methods. In #608 we added support for this. However, there could be cases when we want have Arquillian ignore those parameters. For those cases we should we should add parameter, something like `@JunitParam` or `@ArquillianIgnore` which will allow other `ParameterResolver`'s to resolve the method parameter.

**Comment by rhusar** [comment]
Theoretically, if there were to be multiple `ParameterResolver`s apart from new Arquillian one and the existing JUnit one, the `@JunitParam` wouldn't be exactly correct. 
IIUC the annotation is meant to "do not use arquillian parameter resolution for this parameter", so perhaps something like `@SkipArquillianParameterResolution` or `@SkipArquillianParameterResolver? :-)

**Comment by jamezp** [comment]
Yes, that's where I got `@ArquillianIgnore` from. I was trying to think of something somewhat short.

**Comment by rhusar** [comment]
> Yes, that's where I got `@ArquillianIgnore` from. I was trying to think of something somewhat short.

I am afraid it cannot be short without compromising on clarity since stuff like `ArquillianIgnore` can easily be mistaken for being something else. Sounds to me it should be at least 3 words - `@Ignore/Skip-Arquillian-Resolution/Resolver/Parameter`

**Comment by hantsy** [comment]
The `JunitParam` will be applied on the original Junit parameter injection?

How to keep the compatible with the existing testing libs that already used the JUnit param injection. If we need an explicit qulifier for our Arquillian param injection, I would like to add the annotation to the Arquillian method params. eg. `@ArquillianParam`, it also can be used for CDI scan to filter the method injections.

**Comment by jamezp** [comment]
@hantsy Sorry for the delayed response. This is only an issue if there is a conflict to a parameter both JUnit and Arquillian can produce. For example, if you created a JUnit `ParameterResolver` for  `URI`, Arquillian might be the one injecting the `URI` instead of your custom `ParameterResolver`.

What an introduced annotation would do is act like a CDI `@Vetoed` annotation where Arquillian would not attempt to inject the parameter.

**Comment by hantsy** [comment]
> What an introduced annotation would do is act like a CDI @Vetoed annotation where Arquillian would not attempt to inject the parameter.

I hope this annotation to help Arquillian to recognize/scan the Arquillian specific paramenter injection, and do not affect the basic JUnit5 parameter inject.

**Comment by hantsy** [comment]
It seems using the existing `@ArquillianResource` is a good term. 

I noticed there are somewhere describe it already got support(I do not remember), but I tried in my example project when injecting the Page Object using it with Graphene Driver to initialize the web pages, failed.

**Comment by hantsy** [comment]
Any update on this? I think using the existing `@ArquillianResource` for page object injection is good.

**Comment by rhusar** [comment]
`@ArquillianResource SomePage page` resolves through `ArquillianResourceTestEnricher`, which needs a `ResourceProvider` whose `canProvide()` matches the page type; with none registered it throws `IllegalArgumentException: No ResourceProvider found for type: ...`. That provider would have to live in Graphene, not in core. Could you open an issue against arquillian-graphene with the stack trace and link it here? 

I am still a little confused by @hantsy 's proposal -- changing things to be opt-in now would break a LOT of existing code, so not a real path forward. Given the releases and exposure of the current way of functioning, it seems to me the only path forward is what is being proposed, i.e. the ignoring.

**Comment by jamezp** [comment]
Yes. I don't think Arquillian Core should have a separate annotation. If any projects that build onto Arquillian and don't want to use a test enricher, they can have their own annotations.

**Comment by hantsy** [comment]
@rhusar The ArquillianResource is only used to inject the deployment URL?

**Comment by rhusar** [comment]
@hantsy no, that's not right, for fields nor method parameters. @ArquillianResource is a generic lookup against the ResourceProvider SPI; URL is just the most common provider.

Mechanically speaking, method-parameter injection is the same path as field injection – same lookup method.

Just an example, list of providers registered in a [WildFly](https://github.com/wildfly/wildfly) testsuite runs:

* URL, URI
* Deployer
* ContainerController
* InitialContext
* ManagementClient
* ServerManager
* MBeanServerConnection
* ServiceContainer
* ServiceTarget

What we use only in parameter injections is smaller set:

* URL
* InitialContext
* ManagementClient
* Deployer
